src/com/goide/inspections/GoExportedOwnDeclarationInspection.java

```diff
@@ -0,0 +1,207 @@
+    return new GoVisitor() {
+      @Override
+      public void visitConstDeclaration(@NotNull GoConstDeclaration o) {
+        if (!(o.getParent() instanceof GoFile)) {
+          return;
+        }
+
+        for (int i = 0; i < o.getConstSpecList().size(); i++) {
+          int index = 0;
+          for (GoConstDefinition constDefinition : o.getConstSpecList().get(i).getConstDefinitionList()) {
+            checkElement(holder, "const", constDefinition, index);
+            index++;
+          }
+        }
+      }
+
+      @Override
+      public void visitVarDeclaration(@NotNull GoVarDeclaration o) {
+        if (!(o.getParent() instanceof GoFile)) {
+          return;
+        }
+
+        for (int i = 0; i < o.getVarSpecList().size(); i++) {
+          int index = 0;
+          for (GoVarDefinition varDefinition : o.getVarSpecList().get(i).getVarDefinitionList()) {
+            checkElement(holder, "var", varDefinition, index);
+            index++;
+          }
+        }
+      }
+    };
+  }
+
+  public static void checkElement(ProblemsHolder holder, String type, GoNamedElement namedElement, int index) {
+    if (index < 1 || !namedElement.isPublic()) {
+      return;
+    }
+
+    String errorText = "Exported " + type + " '" + namedElement.getName() + "' should have its own declaration";
+    holder.registerProblem(namedElement, errorText, ProblemHighlightType.WEAK_WARNING, new MyLocalQuickFixBase(index));
+  }
+
+  private static class MyLocalQuickFixBase extends LocalQuickFixBase {
+    private int index;
+
+    public MyLocalQuickFixBase(int index) {
+      super(QUICK_FIX_NAME);
+      this.index = index;
+    }
+
+    @Override
+    public void applyFix(@NotNull final Project project, @NotNull ProblemDescriptor descriptor) {
+      final PsiElement element = descriptor.getPsiElement();
+      if (element == null) return;
+      new WriteCommandAction.Simple(project, getName(), element.getContainingFile()) {
+        @Override
+        protected void run() throws Throwable {
+          List<GoExpression> elementValueList;
+          String elementType = "";
+          if (element instanceof GoConstDefinition) {
+            elementValueList = ((GoConstSpec)element.getParent()).getExpressionList();
+            GoType elemType = ((GoConstSpec)element.getParent()).getType();
+            if (elemType != null) {
+              elementType = elemType.getText();
+            }
+          }
+          else if (element instanceof GoVarDefinition) {
+            elementValueList = ((GoVarSpec)element.getParent()).getExpressionList();
+            GoType elemType = ((GoVarSpec)element.getParent()).getType();
+            if (elemType != null) {
+              elementType = elemType.getText();
+            }
+          }
+          else {
+            return;
+          }
+
+          if (elementValueList.size() != 0 &&
+              elementValueList.size() <= index) {
```

Comment:
Actually, at this moment index might be different already, you have to recalculate it.